### ISSUE TITLE
[Cleanup] Create a worker for location of interest sync

### ISSUE THREAD

**Issue by shobhitagarwal1612** [google/ground-android#988]
Constructor of `MainViewModel` is doing some actual work which is not a [recommended](http://misko.hevery.com/code-reviewers-guide/flaw-constructor-does-real-work/#:~:text=Object%20graph%20creation%20is%20a,violates%20the%20Single%20Responsibility%20Principle.&text=Testing%20such%20constructors%20is%20difficult,object%2C%20the%20constructor%20must%20execute.) pattern. 
On top of that, it also creates a problem while unit testing the class as we need some extra configuration to ignore the logic in the constructor.

https://github.com/google/ground-android/blob/92cb94d5d7c54fe1e9548814ff9df21d0af33edd/gnd/src/main/java/com/google/android/gnd/MainViewModel.java#L89-L94

The feature syncing should be done using a worker similar to how the mutations are synced.

**Comment by gino-m** [comment]
Agreed. The right way to do this would be to create a long-running worker, which requires a notification UI to be shown while it's running in the background (similar to what media players do on Android, for example).

**Comment by dugoalberto** [comment]
It's my first issue, I would like to take it on, is this possible?

**Comment by gino-m** [comment]
Related to #1125

**Comment by nikhil6624** [comment]
Hi @shobhitagarwal1612 , May I work on this issue?

**Comment by shobhitagarwal1612** [comment]
Hey @nikhil6624! We are currently migrating our codebase to kotlin and have paused feature development at the moment. Would you like to contribute to that?

**Comment by nikhil6624** [comment]
Sure @shobhitagarwal1612 , I would be more than happy to do that! You can allocate me a task/tasks of your choice. I'll give my best to it. :)

**Comment by shobhitagarwal1612** [comment]
This is the common bug for tracking kotling migrations https://github.com/google/ground-android/issues/532

You can pick the following directory which has 3 java classes and try converting to kotlin.
https://github.com/google/ground-android/tree/master/ground/src/main/java/com/google/android/ground/persistence/local/room/relations

**Comment by nikhil6624** [comment]
Sure. Will start working on these. Thanks for a swift response @shobhitagarwal1612    :)

**Comment by shobhitagarwal1612** [comment]
Please make sure that you've gone through the [contribution guidelines](https://github.com/google/ground-android/blob/master/CONTRIBUTING.md) before starting with anything. Welcome to the project!

**Comment by nikhil6624** [comment]
I have submitted the CLA by visiting https://cla.developers.google.com/ beforehand. Thanks for inducting me @shobhitagarwal1612  :)

**Comment by gino-m** [comment]
This issue is actually quite complex, since it will require Cloud Messaging to wake up the worker when remote changes have been made.  Closing in favor of #1373.
